DFW.cpp

```diff
@@ -10,298 +10,325 @@
+				 }else{
+				 Serial.print("  ");
+				 Serial.print(packet[i], HEX);
+				 }
+				 }
+				 */
+			}
+			lastHeartBeatTime = millis();
+		}
 	}
 
 	//Heartbeat need to read once every 1 seconds
-	hbTime = millis() - lastTime;
+	hbTime = millis() - lastHeartBeatTime;
 	if (hbTime > 1000) {
-		// Serial.println("HB tripped");
-		//Serial.println("HB");
+		Serial.print("\r\nHB tripped clearing data. Diff= ");
+		Serial.print(hbTime);
+		Serial.print(" lastTime = ");
+		Serial.print(lastHeartBeatTime);
+		Serial.print(" current Time = ");
+		Serial.print(millis());
 		digitalWrite(debuginpin, 1);
 		for (int i = 0; i < 2; i++) {
 			byteBu[i] = 127;
 		}
-		for (int i = 0; i <= 4; i++) {
+		for (int i = 0; i < 5; i++) {
 			byteAn[i] = 90;
 		}
+
+	}else{
+		long timeSinceFlash = millis() - flashTime;
+		if(timeSinceFlash>1000){
+			flashTime=millis();
+			digitalWrite(debuginpin, 1);
+		}else{
+			digitalWrite(debuginpin,0);
+		}
 	}
+}
 
+DFW::DFW(int debugpin, void (*autonomous)(long), void (*teleop)(long)) //13 is easiest
+		{
+	pinMode(debugpin, OUTPUT);
+	digitalWrite(debugpin, 0);
+	debuginpin = debugpin;
+	myAutonomous = autonomous;
+	myTeleop = teleop;
+	state = powerup;
+	//printing = false;
 }
 
 DFW::DFW(int debugpin)    //13 is easiest
 		{
 	pinMode(debugpin, OUTPUT);
 	digitalWrite(debugpin, 0);
 	debuginpin = debugpin;
+	myAutonomous = Null;
+	myTeleop = Null;
+	state = powerup;
+	//printing = false;
 }
-
-void DFW::begin(long baud, int port)    //serial1 is the only functional input
+CompetitionState DFW::getCompetitionState(void) {
+	return state;
+}
+void DFW::begin(long baud)    //serial1 is the only functional input
 		{
+	Serial1.begin(baud);
+	//Serial1.setTimeout(250);
 
-	switch (port) {
-	case 0:
-		Serial.begin(baud);
-		break;
-	case 1:
-		Serial1.begin(baud);
-		Serial1.setTimeout(250);
-		break;
-	case 2:
-		Serial2.begin(baud);
-		break;
-	case 3:
-		Serial3.begin(baud);
-		break;
-	default:
-		Serial1.begin(baud);
-		break;
-	}
 }
 
 int DFW::joystickrv(void) // returns joystick right vertical
 		{
 //Serial.println("jrv");
-	unsigned char stickRv = byteAn[1];
+	unsigned char stickRv = byteAn[0];
 	return (stickRv);
 }
 
 int DFW::joystickrh(void) // returns joystick right vertical
 		{
-	unsigned char stickrh = byteAn[2];
+	unsigned char stickrh = byteAn[1];
 	return (stickrh);
 }
 
 int DFW::joysticklv(void) // returns joystick right vertical
 		{
-	unsigned char sticklv = byteAn[3];
+	unsigned char sticklv = byteAn[2];
 	return (sticklv);
 }
 
 int DFW::joysticklh(void) // returns joystick right vertical
 		{
-	unsigned char sticklh = byteAn[4];
+	unsigned char sticklh = byteAn[3];
 	return (sticklh);
 }
 
-int DFW::start(void) // returns start button state 
+bool DFW::start(void) // returns start button state
 		{
-	int startB;
-	if (((byteBu[1] & 0b00000010) == 0b00000010)) {
-		startB = 1;
-	} else {
-		startB = 0;
+	if ((byteBu[1] & 0b00000010) != 0) {
+		return false;
 	}
-	return (startB);
+	return true;

```

Comment:
Can we return the comparison directly? i.e.: `return (byteBu[1] & 0b00000010) == 0`